janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphBlueprintsTransaction.java

```diff
@@ -201,10 +201,10 @@ public boolean isOpen() {
             }
 
             @Override
-            public void doClose() {
-                getGraph().tinkerpopTxContainer.close(this);
-
-                // calling super will clear listeners
+            protected void doClose() {
+                if (isOpen()) {
+                    throw Exceptions.openTransactionsOnClose();
+                }
```

Comment:
Hey @mad ,
this piece of code is giving me problems when trying to refactor some Janus code.

Could you please explain to me why you throw exception if the tx is open?
The exception only says "Commit or rollback all outstanding transactions before closing the transaction", which does not really make sense.

Reply:
Do you think we could just close the JanusGraphBlueprintsTransaction? something like:
```
 protected void doClose(){
        JanusGraphBlueprintsTransaction.this.close();
        super.doClose();
 }

Reply:
Hi @marco-scoppetta 

This is MANUAL mode transaction, see javadoc for `JanusGraph.newTransaction()` and `AbstractTransaction.createThreadedTx`, it needs to commit or rolled backed explicitly

More readable code may looks like this:
```
            @Override
            protected void doClose() {
                CLOSE_BEHAVIOR.MANUAL.accept(this);
                super.doClose();
            }

```

Reply:
I see, thanks @mad ,

but do you know why the check on `isOpen()`?
In my reasoning if I want to close a transaction, I would naturally except it to be Open before closing it, but looking at this code it looks like we expect something to be already closed when we close the Transaction.
I am sure I am missing something here, so wondering if you could shed some light on the topic.

Thank you

Reply:
@marco-scoppetta tx closed when commited or rolled backed

```
try (tx) {
}
```
Should return exception, if used manual-mode transaction, because tx.commit()/tx.rollback()

Reply:
Oh I see, you're saying that a manual transaction, such as the one built using `graph.newTransaction()` should either be `rollback`ed or `commit`ed, when trying to close it without rollback or commit should throw exception!

Reply:
Yes